code/go/0chain.net/blobbercore/challenge/challenge.go

```diff
@@ -215,33 +281,91 @@ func commitProcessed(ctx context.Context) {
 	}()
 
 	db := datastore.GetStore().GetDB()
-	var challenges []*ChallengeEntity
+	count := 0
 
-	db.Where(ChallengeEntity{Status: Processed}).
-		Order("sequence").
-		Find(&challenges)
+	rows, err := db.Model(&ChallengeEntity{}).
+		Where("status = ?", Processed).
+		Select("challenge_id", "created_at").Rows()
 
-	if len(challenges) > 0 {
+	if err != nil {
+		logging.Logger.Error("[challenge]validate: ",
+			zap.Error(err))
+		return
+	}
+	defer rows.Close()
+
+	startTime := time.Now()
+	swg := sizedwaitgroup.New(config.Configuration.ChallengeResolveNumWorkers)
 
-		startTime := time.Now()
+	for rows.Next() {
+		count++
+		now := time.Now()
 
-		swg := sizedwaitgroup.New(config.Configuration.ChallengeResolveNumWorkers)
-		for _, challenge := range challenges {
-			swg.Add()
-			go func(c *ChallengeEntity) {
-				defer swg.Done()
-				commitChallenge(c)
-			}(challenge)
+		var challengeID string
+		var createdAt time.Time
+
+		err := rows.Scan(&challengeID, &createdAt)
+		if err != nil {
+			logging.Logger.Error("[challenge]commit: ",
+				zap.Error(err))
+			continue
+		}
+
+		if time.Since(createdAt) > config.Configuration.ChallengeCompletionTime {
```

Comment:
if this created at is 0chain's response challenge time, then we might have an issue computing difference

Reply:
it should be ok. because @guruhubb  asked devops to set all server with UTC.